File: JSPM.Web/Modules/ServerTypes/Master/StatesRow.ts
import { fieldsProxy } from "@serenity-is/corelib/q";

export interface StatesRow {
    Id?: number;
    Name?: string;
}

export abstract class StatesRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'Name';
    static readonly localTextPrefix = 'Master.States';
    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<StatesRow>();
}

File: JSPM.Web/Modules/Master/States/StatesRow.cs
﻿using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace JSPM.Master;

[ConnectionKey("Default"), Module("Master"), TableName("[Master].[States]")]
[DisplayName("States"), InstanceName("States")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
public sealed class StatesRow : Row<StatesRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Id"), Identity, IdProperty]
    public int? Id
    {
        get => fields.Id[this];
        set => fields.Id[this] = value;
    }

    [DisplayName("Name"), Size(100), NotNull, QuickSearch, NameProperty]
    public string Name
    {
        get => fields.Name[this];
        set => fields.Name[this] = value;
    }

    public class RowFields : RowFieldsBase
    {
        public Int32Field Id;
        public StringField Name;

    }
}